docs/source/driver/go/flight_sql.rst

```diff
@@ -80,63 +84,58 @@ API specification 1.0.0, as well as some additional, custom options.
 Authentication
 --------------
 
-The driver does no authentication by default.
+The driver does no authentication by default.  The driver implements a
+few optional authentication schemes:
 
-The driver implements one optional authentication scheme that mimics
-the Arrow Flight SQL JDBC driver.  This can be enabled by setting the
-option ``arrow.flight.sql.authorization_header`` on the
-:cpp:class:`AdbcDatabase`.  The client provides credentials by setting
-the option value to the value of the ``authorization`` header sent
-from client to server.  The server then responds with an
-``authorization`` header on the first request.  The value of this
-header will then be sent back as the ``authorization`` header on all
-future requests.
+- Mutual TLS (mTLS): see "Client Options" below.
+- An HTTP-style scheme mimicking the Arrow Flight SQL JDBC driver.
+
+  Set the options ``username`` and ``password`` on the
+  :cpp:class:`AdbcDatabase`.  Alternatively, set the option
+  ``arrow.flight.sql.authorization_header`` for full control.
```

Comment:
I'm not seeing this implemented in the Go code, does it still need to be done or did I miss it?

Reply:
Not yet, I plan to tackle these over the course of several PRs (there are a lot of things in these docs that are not yet implemented)